src/nrnmpi/bbsmpipack.cpp

```diff
@@ -20,18 +20,7 @@
 #include <nrnmpi_impl.h>
 #include <hocdec.h>
 
-#if 0
-#define guard(f) nrn_assert(f == MPI_SUCCESS)
-#else
-#define guard(f)                       \
```

Comment:
It is an interesting question, to which I don't know the answer, whether one can recover from a collective error or should just abort (or assert(0) or MPI_ABORT) A perennial problem is wasting the entire job time by blocking when there is an early failure. nrn_timeout is supposed to alleviate this somewhat, but a never realized goal was to give a reason for the timeout.

Reply:
Should we change nothing for this file?

Reply:
> whether one can recover from a collective error or should just abort (or assert(0) or MPI_ABORT) A perennial problem is wasting the entire job time by blocking when there is an early failure

One can create custom error handlers using [MPI_Comm_create_errhandler](https://www.mpich.org/static/docs/v3.1/www3/MPI_Comm_create_errhandler.html) or [MPI_Errhandler_create](https://www.mpich.org/static/docs/v3.3/www3/MPI_Errhandler_create.html) but I guess we need to do `MPI_Abort` anyway. So it depends if we want to do something specific via custom error handler.